File: src/main/kotlin/eu/alkismavridis/bambumbot/entities/BambumRobot.kt
package eu.alkismavridis.bambumbot.entities

import eu.alkismavridis.bambumbot.entities.brain_structure.BrainMetadata
import eu.alkismavridis.bambumbot.entities.brain_structure.BrainConfig
import eu.alkismavridis.bambumbot.entities.data_provider.DataProviderConfig
import eu.alkismavridis.bambumbot.entities.train.TrainerConfig
import java.util.*

class BambumRobot(
        val id: UUID,
        val name: String,
        val brainConfig: BrainConfig,
        val brainMetadata: BrainMetadata,
        val dataProviderConfig: DataProviderConfig,
        val trainerConfig: TrainerConfig,

        val bestState: DoubleArray,
        var leastError: Double,

        val currentState: DoubleArray
) {}

File: frontend/src/model/BambumRobot.ts
import BrainMetadata from "./BrainMetadata";
import TrainingData from "./TrainingData";
import TrainerConfig from "./TrainerConfig";
import DataProviderConfig from "./DataProviderConfig";

export default class BambumRobot {
    id: string = "";
    name: string = "";
    brainMetadata: BrainMetadata;

    dataProviderConfig: DataProviderConfig;
    trainerConfig: TrainerConfig;

    bestState: number[];
    leastError: number = 9999999999;
    currentState: number[];

    trainingData: TrainingData;
}
